src/jit/lclvars.cpp

```diff
@@ -2321,6 +2321,30 @@ void Compiler::lvaSetStruct(unsigned varNum, CORINFO_CLASS_HANDLE typeHnd, bool
         unsigned  numGCVars;
         var_types simdBaseType = TYP_UNKNOWN;
         varDsc->lvType         = impNormStructType(typeHnd, varDsc->lvGcLayout, &numGCVars, &simdBaseType);
+        
+#if (defined(_TARGET_AMD64_) && defined(_TARGET_WINDOWS_)) || defined(_TARGET_ARM64_)
+        if (varDsc->lvIsParam)
+        {
+            varDsc->lvIsImplicitByRef = 0;
+
+            // Mark implicit byrefs
+            unsigned structSize = varDsc->lvSize();
+
+            if (structSize <= REGSIZE_BYTES) 
+            {
+                structSize = varDsc->lvExactSize;
+            }
+
+    #if defined(_TARGET_AMD64_)
+            if (structSize > REGSIZE_BYTES || (structSize & (structSize - 1)) != 0)
+    #else // defined(_TARGET_ARM64_)
+            if (structSize > TARGET_POINTER_SIZE && !lvaIsMultiregStruct(varDsc, this->info.compIsVarArgs))
+    #endif // defined(_TARGET_AMD64_)
+            {
+                varDsc->lvIsImplicitByRef = 1;
+            }
+        }
+#endif // (defined(_TARGET_AMD64_) && defined(_TARGET_WINDOWS_)) || defined(_TARGET_ARM64_)
```

Comment:
Should this instead call `getArgTypeForStruct`?

Reply:
Done